src/Features/LanguageServer/Protocol/RoslynRequestExecutionQueue.cs

```diff
@@ -27,5 +32,23 @@ public override Task WrapStartRequestTaskAsync(Task nonMutatingRequestTask, bool
                 return nonMutatingRequestTask.ReportNonFatalErrorAsync();
             }
         }
+
+        private void TrySetLocale()
+        {
+            var locale = _initializeManager.TryGetInitializeParams()?.Locale;
+            // The client may not have given us a UI or this is the initialize request and we haven't saved it yet.
+            if (!string.IsNullOrWhiteSpace(locale))
+            {
+                try
+                {
+                    var desiredUICulture = CultureInfo.CreateSpecificCulture(locale);
+                    CultureInfo.CurrentUICulture = desiredUICulture;
```

Comment:
we set the UI for the request here.  Since we set it right before actually running the request, the desired culture will flow down the stack and will be respected by the actual request.

Reply:
It actually does, but that is because the LSP spec defines the locale in the initializeparams.  If it was defined somewhere else though we could technically support it on the server side.